trin-core/src/portalnet/overlay_service.rs

```diff
@@ -1227,17 +1224,17 @@ where
         self.utp_listener_tx
             .send(utp_request).map_err(|err| anyhow!("Unable to send Connect request to UtpListener when processing ACCEPT message: {err}"))?;
 
-        let mut conn = rx.await?;
-        // Handle STATE packet for SYN
-        let mut buf = [0; BUF_SIZE];
-        conn.recv(&mut buf).await?;
-
         let content_items = self.provide_requested_content(&response, content_keys_offered)?;
 
         let content_payload = ContentPayloadList::new(content_items)
             .map_err(|err| anyhow!("Unable to build content payload: {err:?}"))?;
 
         tokio::spawn(async move {
+            let mut conn = rx.await.unwrap();
+            // Handle STATE packet for SYN
+            let mut buf = [0; BUF_SIZE];
+            conn.recv(&mut buf).await.unwrap();
```

Comment:
@ogenev Are there any consequences I'm missing from moving this login into the spawned task?

Reply:
It should be fine but I think we have to call `provide_requested_content` after we initiate the uTP handshake. If we can't establish an uTP connection with the remote node, it doesn't make sense to try to get the requested content from the db. 

So I will move those lines after `conn.recv(&mut buf).await.unwrap()` :

```rust
let content_items = self.provide_requested_content(&response, content_keys_offered)?;

let content_payload = ContentPayloadList::new(content_items)
     .map_err(|err| anyhow!("Unable to build content payload: {err:?}"))?;
```